Docker buildx bake panics with empty variable in JSON file

```
docker buildx version   
github.com/docker/buildx v0.7.1-docker 05846896d149da05f3d6fd1e7770da187b52a247
```

For this JSON:
```json
{
  "variable": {
    "TAG": {
    }
  },
  "group": {
    "default": {
      "targets": ["webapp"]
    }
  },
  "target": {
    "webapp": {
      "tags": ["docker.io/username/webapp:${TAG}"],
      "dockerfile": "aa"
    }
  }
}
```

I get the following panic:

```
 TAG=ss docker buildx bake -f a.json  ⬡
[+] Building 0.0s (0/0)                                                                                                                                                                                                                                                                                                                                                             
panic: reflect: call of reflect.Value.IsZero on zero Value

goroutine 1 [running]:
reflect.Value.IsZero({0x0, 0x0, 0xc00059afc0})
        reflect/value.go:1475 +0x27f
github.com/docker/buildx/bake/hclparser.appendJSONFuncCalls({0x557d25c1a7b8, 0xc000618250}, 0x557d25c1a7b8)
        github.com/docker/buildx/bake/hclparser/expr.go:86 +0x41f
github.com/docker/buildx/bake/hclparser.jsonFuncCallsRecursive({0x557d25c1a7b8, 0xc0006181f0})
        github.com/docker/buildx/bake/hclparser/expr.go:50 +0x196
github.com/docker/buildx/bake/hclparser.funcCalls({0x557d25c1a7b8, 0xc0006181f0})
        github.com/docker/buildx/bake/hclparser/expr.go:15 +0x13f
github.com/docker/buildx/bake/hclparser.(*parser).loadDeps(0x557d25904b00, {0x557d25c1a7b8, 0xc0006181f0}, 0x8)
        github.com/docker/buildx/bake/hclparser/hclparser.go:58 +0x54
github.com/docker/buildx/bake/hclparser.(*parser).resolveValue(0xc00004f9a0, {0xc0007dcbb0, 0x8})
        github.com/docker/buildx/bake/hclparser/hclparser.go:200 +0x23c
github.com/docker/buildx/bake/hclparser.Parse({0x557d25c1b350, 0xc00069e660}, {0x557d25bc7900, 0xc00088ae10}, {0x557d259fcb20, 0xc00088ae40})
        github.com/docker/buildx/bake/hclparser/hclparser.go:318 +0x80d
github.com/docker/buildx/bake.ParseFiles({0xc00088ade0, 0x1, 0x5}, 0xc000700000)
        github.com/docker/buildx/bake/bake.go:140 +0x14d
github.com/docker/buildx/bake.ReadTargets({0x13, 0x0}, {0xc00088ade0, 0xc000056064, 0x2a}, {0xc000713a70, 0x1, 0x1}, {0x0, 0x0, ...}, ...)
        github.com/docker/buildx/bake/bake.go:71 +0x5a
github.com/docker/buildx/commands.runBake({0x557d25c67298, 0xc000434000}, {0xc00031c3a0, 0x0, 0x2}, {{0xc0003d84e0, 0x1, 0x1}, {0x0, 0x0, ...}, ...})
        github.com/docker/buildx/commands/bake.go:107 +0xb06
github.com/docker/buildx/commands.bakeCmd.func1(0xc000149400, {0xc00031c3a0, 0x0, 0x2})
        github.com/docker/buildx/commands/bake.go:187 +0x1e8
github.com/spf13/cobra.(*Command).execute(0xc000149400, {0xc00080c220, 0x2, 0x2})
        github.com/spf13/cobra@v1.2.1/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc000906280)
        github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.2.1/command.go:902
github.com/docker/cli/cli-plugins/plugin.RunPlugin(0x0, 0xc000148280, {{0x557d2545047e, 0x5}, {0x557d25459b5a, 0xb}, {0x557d256f4e68, 0xd}, {0x0, 0x0}, ...})
        github.com/docker/cli@v20.10.8+incompatible/cli-plugins/plugin/plugin.go:51 +0x130
main.main()
        github.com/docker/buildx/cmd/buildx/main.go:77 +0x325
```

--- a/bake/hcl_test.go
+++ b/bake/hcl_test.go
@@ -745,3 +745,13 @@ target "two" {
 	require.Equal(t, c.Targets[1].Name, "two")
 	require.Equal(t, map[string]string{"b": "pre-jkl"}, c.Targets[1].Args)
 }
+
+func TestEmptyVariableJSON(t *testing.T) {
+	dt := []byte(`{
+	  "variable": {
+	    "VAR": {}
+	  }
+	}`)
+	_, err := ParseFile(dt, "docker-bake.json")
+	require.NoError(t, err)
+}

--- a/bake/hclparser/hclparser.go
+++ b/bake/hclparser/hclparser.go
@@ -256,6 +256,7 @@ func Parse(b hcl.Body, opt Opt, val interface{}) hcl.Diagnostics {
 	if err := gohcl.DecodeBody(b, nil, &defs); err != nil {
 		return err
 	}
+	defsSchema, _ := gohcl.ImpliedBodySchema(defs)
 
 	if opt.LookupVar == nil {
 		opt.LookupVar = func(string) (string, bool) {
@@ -298,6 +299,16 @@ func Parse(b hcl.Body, opt Opt, val interface{}) hcl.Diagnostics {
 			continue
 		}
 		p.funcs[v.Name] = v
+	}
+
+	content, b, diags := b.PartialContent(schema)
+	if diags.HasErrors() {
+		return diags
+	}
+
+	_, b, diags = b.PartialContent(defsSchema)
+	if diags.HasErrors() {
+		return diags
 	}
 
 	attrs, diags := b.JustAttributes()
@@ -369,11 +380,6 @@ func Parse(b hcl.Body, opt Opt, val interface{}) hcl.Diagnostics {
 				},
 			}
 		}
-	}
-
-	content, _, diags := b.PartialContent(schema)
-	if diags.HasErrors() {
-		return diags
 	}
 
 	for _, a := range content.Attributes {

